conans/test/functional/package_id/package_id_test.py

```diff
@@ -78,3 +79,40 @@ def package(self):
         client.run("install test/0.1@danimtb/testing")
         client.run("search test/0.1@danimtb/testing")
         self.assertIn("compiler.version: kk=kk", client.out)
+
+    def option_in_test(self):
+        # https://github.com/conan-io/conan/issues/7299
+        conanfile = textwrap.dedent("""
+            from conans import ConanFile
+
+            class TestConan(ConanFile):
+                options = {"fpic": [True, False]}
+                default_options = {"fpic": True}
+                def package_id(self):
+                    if "fpic" in self.options:
+                        self.output.info("fpic is an option!!!")
+                    if "fpic" in self.info.options:
+                        self.output.info("fpic is an info.option!!!")
+                    if "other" not in self.options:
+                        self.output.info("other is not an option!!!")
+                    if "other" not in self.info.options:
```

Comment:
Add the comment, so we are free to modify it in the future.

```suggestion
                    if "other" not in self.info.options:  # Not documented
```